types/src/transaction/transaction.rs

```diff
@@ -264,6 +232,146 @@ fn is_input_and_output_consistent(inputs: &[AssetTransferInput], outputs: &[Asse
     sum.iter().all(|(_, sum)| sum.is_zero())
 }
 
+fn apply_bitmask(
+    mut bitmask: u128,
+    outputs: Vec<AssetTransferOutput>,
+    mut result: Vec<AssetTransferOutput>,
+) -> Result<Vec<AssetTransferOutput>, HashingError> {
+    for i in 0.. {
+        if bitmask == 0 {
+            return Ok(result)
+        }
+
+        if outputs.is_empty() {
+            return Err(HashingError::InvalidFilter)
+        }
+
+        if (bitmask & 0x1) == 1 {
+            result.push(outputs[i].clone());
+        }
+        bitmask = bitmask >> 1
+    }
+    Err(HashingError::InvalidFilter)
+}
+
+impl PartialHashing for Transaction {
+    fn hash_partially(&self, bitmask: u128, cur: &AssetOutPoint, burn: bool) -> Result<H256, HashingError> {
+        let input_scheme = (bitmask & 0x1) == 1;
+        let output_scheme = (bitmask >> 1 & 0x1) == 1;
+        let mask = bitmask >> 2;
+
+        match self {
+            Transaction::AssetTransfer {
+                network_id,
+                burns,
+                inputs,
+                outputs,
+                nonce,
+            } => {
+                let new_burns = if input_scheme {
```

Comment:
It seems that 271~303 and 305~337 do the same thing.
Please extract it a function.

Reply:
I resolved it